src/com/oltpbenchmark/DBWorkload.java

```diff
@@ -976,44 +977,65 @@ private static void PrintLatencies(List<Worker> workers,
       for (LatencyRecord.Sample sample : w.getLatencyRecords()) {
         list_latencies.get(sample.tranType - 1).add(sample.operationLatencyUs);
       }
-
-      if (displayEnhancedLatencyMetrics) {
-        for (LatencyRecord.Sample sample : w.getWholeOperationLatencyRecords()) {
-          list_enhanced_latencies.get(sample.tranType - 1).add(sample.operationLatencyUs);
-        }
-      }
     }
 
-    if (InstrumentedPreparedStatement.isTrackingLatencyMetrics()) {
-      NewOrder.printLatencyStats();
-      Payment.printLatencyStats();
-      OrderStatus.printLatencyStats();
-      Delivery.printLatencyStats();
-      StockLevel.printLatencyStats();
+    StringBuilder resultOut = new StringBuilder();
+    resultOut.append("\n");
+    resultOut.append("================LATENCIES===============\n");
+    resultOut.append(" Operation  | Avg. Latency | P99 Latency\n");
+    for (int i = 0; i < list_latencies.size(); ++i) {
+      String op = transactionTypes.get(i + 1);
+      List<Integer> latencies = list_latencies.get(i);
+      resultOut.append(String.format(
+          "%11s |%13.2f |%12.2f\n", op, getAverageLatency(latencies), getP99Latency(latencies)));
     }
+    LOG.info(resultOut.toString());
+  }
 
-    if (!displayEnhancedLatencyMetrics) {
-      for (int i = 0; i < list_latencies.size(); ++i) {
-        LOG.info(getOperationLatencyString(transactionTypes.get(i+1),
-                                           list_latencies.get(i)));
+  private static void PrintQueryAttempts(List<Worker> workers, WorkloadConfiguration workConf) {
+    int numTxnTypes = workConf.getNumTxnTypes();
+    int numTriesPerProc = workConf.getMaxRetriesPerTransaction() + 1;
+    int[][] totalRetries = new int[numTxnTypes][numTriesPerProc];
+    int[] totalFailures = new int[numTxnTypes];
+    for (Worker w : workers) {
+      for (int i = 0; i < numTxnTypes; ++i) {
+        totalFailures[i] += w.getTotalFailures()[i];
+        for (int tryIdx = 0; tryIdx < numTriesPerProc; ++tryIdx) {
```

Comment:
Isn't tryIdx = 0 the one where we try the query for the first time..
Do we need to show them?

Reply:
right.